### PR TITLE
Neck Slot

### PR DESCRIPTION
**Pull request by ghost**
OK listen.  I have spent the last 3 and a half hours fucking with this and finally getting everything done after restarting.  I cannot find the place where to get the goddamned neck slot thing to go away when the outfit HUD is not up.  It's just always there.  Someone fucking help.  Otherwise this shit adds Neck Slot and capes/bed sheets use neck slot.  I'll be adding ties and some other stuff to Neck Slot but this bug is eluding me and I'm too tired to see it right now.

### CODE REVIEW

**Comment by ghost** [comment]
Wait was vent scrubbers removed?  Da fuck?  Some extra stuff in there I'll figure it out in the morning.

Also I think I need to add the icon to HUD types other than midnight.

**Comment by retrosquid64** [comment]
great, now we can use the slot to strangle someone with a rope

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
make it so you can put a cable coil so you can choke yourself/someone

**Comment by AlpacaGoesCrazy** [comment]
yes, cable coils! make a button in inventory menu (where you strip others) to 'strangle' (like set internals button) when cable coil is in neck slot. and make it more efficient than regular strangling suffocation

**Comment by Partheo** [comment]
Cable stuff is beyond the scope of this PR.

That being said, neckslot is great. Nice work.

**Comment by ghost** [comment]
Shout out to ThatLing for finding the bug.  Fucking top notch.

**Comment by Amelia0010** [comment]
why did you merge testmerge into your branch?

**Comment by ghost** [comment]
I'm retarded that's why.

**Comment by ghost** [comment]
OK I'm done fucking with it unless anyone else has any problems.  I'll be changing ties/etc to use neck in another pull request after I get the mindswap gun done but it won't be this week.  Kinda want to hammer out the main stuff and get it running before adding more to it.

**Comment by ghost** [comment]
OK never mind I found another bug.  Working on that now.

**Comment by ghost** [comment]
Bed sheets currently don't work but I'm working on it.  Unsure what exactly the problem is just yet.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -259,9 +259,7 @@ There are several things that need to be remembered:
 			if(hud_used.inventory_shown)			//if the inventory is open ...
 				wear_neck.screen_loc = ui_neck		//...draw the item in the inventory screen
 			client.screen += wear_neck					//Either way, add the item to the HUD
-		var/t_state = wear_neck.item_state
-		if(!t_state)
-			t_state = wear_neck.icon_state
+		var/t_state = wear_neck.icon_state
 
 		var/image/standing = wear_neck.build_worn_icon(state = t_state, default_layer = NECK_LAYER, default_icon_file = 'icons/mob/neck.dmi')
```
Comment (ThatLing) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Change to

state = t_state.icon_state

And remove line 262

**Comment by ghost** [comment]
OK I think that's finally everything.  Wait shit I gotta fix corgi cloaks.  *sigh*  Back to the grind.

EDIT: I have just now discovered cloaks do NOT work on Ian.  I think that finishes this shit up then.  Merge-ready if no one else has any problems.

**Comment by Time-Green** [comment]
do not forget the collar OwO

**Comment by Firewolf34** [comment]
Please make the Stethoscope wearable in the neck slot. And maybe the doctor could even start out with one around it's neck. The fact that it is currently not wearable (or even capable of being put in the suit slot of a doctor's coat) really grinds my gears

**Comment by ghost** [comment]
I'll be adding more stuff after neck slot gets merged so I can make sure that the base stuff works properly before adding a bunch of stuff to it.

**Comment by ghost** [comment]
Yup no idea.  Shit compiles and works alright on my end even with the new stuff.

**Comment by AsV9** [comment]
Yeah yeah travis we know this is failing already.

**Pull request merged**